flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/VarCharType.java

```diff
@@ -86,6 +89,9 @@ public LogicalType copy(boolean isNullable) {
 
 	@Override
 	public String asSerializableString() {
+		if (length == MAX_LENGTH) {
```

Comment:
Could we have that only in the `summaryString`? Personally I'd prefer to serialize only the actual types and not the shortcuts.

Reply:
I was also unsure about this change. Summary strings should be sufficient, I agree.